src/components/debate/Rebuttal.vue

```diff
@@ -1,254 +1,402 @@
 
-import For from "@/components/debate/For.vue"
-import Against from "@/components/debate/Against.vue"
-import Create from "@/components/debate/Create.vue"
+import { rebuttalService } from "@/services";
+import { config } from "@/configurations";
+import { getSanitizedTime } from "@/helpers";
 export default {
-    name: 'List',
-    components: {For, Against, Create},
-    data(){
-        return{
-            addComment:false,
-            rebuttal: false,
-            fFor: true,
-            fAgainst: true
-        }
+  name: "Rebuttal",
+  components: { Create, Comment },
+  props: ["rebuttalItem", "rebuttalTime", "uuid"],
+  watch: {
+    uuid: function (newVal, oldVal) {
+      
+      this.rebuttals = null;
+      console.log("Prop changed: ", newVal, " | was: ", oldVal);
+      this.loadRebuttals();
     },
-    methods:{
-        toggleComment(){
-            if(this.addComment){
-                this.addComment = false
-            } else {
-                this.addComment = true
-            }
-        },
-
-        toggleRebuttals(){
-             if(this.rebuttal){
-                this.rebuttal = false
-            } else {
-                this.rebuttal = true
-            }
-        },
-
-        filterAll(){
-            this.fFor = true
-            this.fAgainst = true
-        },
-        filterFor(){
-            this.fFor = true
-            this.fAgainst = false
-        },
-        filterAgainst(){
-            this.fFor = false
-            this.fAgainst = true
-        }
-    }
-}
+  },
+  created() {
+    this.loadRebuttals();
+  },
+  data() {
+    return {
+      rebuttals: null,
+      addComment: false,
+      openRebuttal: false,
+      inclination: null,
+    };
+  },
+  computed: {
+    comment: function () {
+      if (this.rebuttalItem) {
+        return this.rebuttalItem.comment;
+      } else {
+        return "";
+      }
+    },
+    profile: function () {
+      if (this.rebuttalItem) {
+        return this.rebuttalItem.profile.user;
+      } else {
+        return "";
+      }
+    },
+    profilePic: function () {
+      if (this.rebuttalItem) {
+        return this.rebuttalItem.profile.display_pic;
+      } else {
+        return "";
+      }
+    },
+  },
+  methods: {
+    loadRebuttals() {
+      const uuid = this.uuid;
+      if (!uuid) {
+        return;
+      }
+      rebuttalService
+        .getRebuttals(uuid)
+        .then((rebuttals) => {
+          this.rebuttals = rebuttals.rebuttals;
+        })
+        .catch(() => {
+          throw config.messagingConfig.messages.error.unknown_error;
+        });
+    },
+
+    sanitizedTime(createdAt) {
+      return getSanitizedTime(createdAt);
+    },
+
+    toggleComment() {
+      this.addComment = !this.addComment;
+    },
+    toggleRebuttals() {
+      this.rebuttals = null;
+      this.openRebuttal = !this.openRebuttal;
+    },
+    filterAll() {
+      this.inclination = null;
+    },
+    filterFor() {
```

Comment:
the best practice for such function is using computed setters instead of methods.: https://vuejs.org/v2/guide/computed.html#Computed-Setter